nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/document/ExtractDocumentText.java

```diff
@@ -0,0 +1,130 @@
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES).build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name(FIELD_SUCCESS)
+            .description("Content extraction successful").build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder().name(FIELD_FAILURE)
+            .description("Content extraction failed").build();
+
+    private List<PropertyDescriptor> descriptors = Collections.unmodifiableList(Arrays.asList(MAX_TEXT_LENGTH));
+    private Set<Relationship> relationships = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(REL_SUCCESS, REL_FAILURE)));
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final int maxTextLength = context.getProperty(MAX_TEXT_LENGTH).evaluateAttributeExpressions(flowFile).asInteger();
+        final String filename = flowFile.getAttribute("filename");
+
+        try {
+            final AtomicReference<String> type = new AtomicReference<>();
+            final AtomicReference<Boolean> exceptionThrown = new AtomicReference<>(false);
+
+            flowFile = session.write(flowFile, (inputStream, outputStream) -> {
+                BufferedInputStream buffStream = new BufferedInputStream(inputStream);
+                Tika tika = new Tika();
+                String text = "";
+                try {
+                    type.set(tika.detect(buffStream, filename));
+                    tika.setMaxStringLength(maxTextLength);
+                    text = tika.parseToString(buffStream);
+
+                } catch (TikaException e) {
+                    getLogger().error("Parsing failed ", e);
+                    exceptionThrown.set(true);
+                }
+
+                outputStream.write(text.getBytes());
+                buffStream.close();
+            });
+
+            if (exceptionThrown.get()) {
+                session.transfer(flowFile, REL_FAILURE);
+            } else {
+
+                Map<String, String> mimeAttrs = new HashMap<>();
+                mimeAttrs.put("mime.type", TEXT_PLAIN);
+                mimeAttrs.put("orig.mime.type", type.get());
+
+                flowFile = session.putAllAttributes(flowFile, mimeAttrs);
+                session.transfer(flowFile, REL_SUCCESS);
+            }
+        } catch (final Throwable t) {
+            getLogger().error("Unable to process ExtractTextProcessor file {} " +
+                    "{} failed to process due to {}; rolling back session", new Object[]{ t.getLocalizedMessage(), this, t});
```

Comment:
This log invocation can be streamlined to avoid duplicating the exception message, and the unnecessary `Object[]` wrapper:
```suggestion
            getLogger().error("Extraction Failed {}", flowFile, t);
```